libs/agent-sm/agent/build.gradle

```diff
@@ -15,6 +15,7 @@ dependencies {
   compileOnly "com.google.code.findbugs:jsr305:3.0.2"
 
   testImplementation "junit:junit:${versions.junit}"
+  testImplementation project(":test:framework")
```

Comment:
Please remove this dependency, the test framework will depend on the agent (see please https://github.com/opensearch-project/OpenSearch/pull/16731 for completeness)